sound/soc/sof/loader.c

```diff
@@ -465,7 +466,7 @@ static void sof_get_windows(struct snd_sof_dev *sdev)
 		}
 	}
 
-	if (outbox_size == 0 || inbox_size == 0) {
+	if (outbox_size == 0 || outbox_size > SOF_IPC_MSG_MAX_SIZE || inbox_size == 0) {
```

Comment:
Interesting, I just looked through the driver code, and we never actually use the `inbox_size`. We use it below to call `snd_sof_dsp_mailbox_init()`, which just records it into `sdev->dsp_box.size` which is then never used. So far we only read fixed-size objects from the DSP: the common message header, trace and (optionally) DMA position updates, but still it seems not perfectly clean... Maybe at least `snd_sof_ipc_msg_data()` should check its `sz` parameter against it.

Reply:
@lyakh . Yes, but this is another store. This PR is for host box. Could you help to add a PR for dsp box ?  Thanks!

Reply:
@lyakh  add another PR to fix it https://github.com/thesofproject/linux/pull/2876